cmd/serverless/main.go

```diff
@@ -86,7 +89,8 @@ func main() {
 	}
 }
 
-func runAgent(tagger tagger.Component) {
+func runAgent(tagger tagger.Component, compression compression.Component) {
```

Comment:
@GustavoCaso Question about this.. 

The current way the `compression.Component` is created requires a [`config.Component`](https://github.com/DataDog/datadog-agent/blob/main/comp/serializer/compression/selector/def.go#L15). Serverless requires the compression component here but doesn't have a `config.Component` - it will call the new method `WithKindAndLevel` that we discussed to get the final compression that it will use.

What's the best solution here?